src/marks/auto.d.ts

```diff
@@ -3,26 +3,208 @@ import type {CompoundMark, Data} from "../mark.js";
 import type {Reducer} from "../reducer.js";
 import type {BinOptions} from "../transforms/bin.js";
 
+/**
+ * One of the categories of mark; *line* includes lineX and lineY; *area*
+ * includes areaX and areaY; *rule* includes ruleX and ruleY; *bar* includes
+ * barX, barY, rectX, and rectY; and *dot* is just dot.
+ */
+type MarkType = "dot" | "line" | "area" | "rule" | "bar";
+
 export interface AutoOptions {
-  x?: ChannelValue | Reducer | ({value?: ChannelValue; reduce?: Reducer | null; zero?: boolean} & BinOptions);
-  y?: ChannelValue | Reducer | ({value?: ChannelValue; reduce?: Reducer | null; zero?: boolean} & BinOptions);
-  color?: ChannelValue | Reducer | {value?: ChannelValue; reduce?: Reducer | null; color?: string};
-  size?: ChannelValue | Reducer | {value?: ChannelValue; reduce?: Reducer | null};
+  /**
+   * The horizontal spatial encoding. If **y** is set without **x**, the **x**
+   * channel will default to the *count* reducer to produce a histogram; you can
+   * override this by passing {reduce: null}.
+   */
+  x?:
+    | ChannelValue
+    | Reducer
+    | ({
+        /**
+         * A string representing a column of your data, or an accessor function.
+         */
```

Comment:
We don’t document these fields e.g. for ChannelValueSpec, so I’m going to fold these comments back up into the option-level documentation. And then it’ll be surfaced earlier when you look at the option, to boot.